Sources/Sentry/SentryScope.m

```diff
@@ -244,6 +249,18 @@ - (void)setLevel:(enum SentryLevel)level
     [self notifyListeners];
 }
 
+- (void)addAttachment:(SentryAttachment *)attachment
+{
+    @synchronized(self) {
+        [self.attachmentArray addObject:attachment];
+    }
+}
+
+- (NSArray<SentryAttachment *> *)attachments
+{
+    return self.attachmentArray.copy;
```

Comment:
I believe we need to get exclusive access here before we copy this

@synchronize(attachmentArray)